src/components/templates/collection-modal/index.tsx

```diff
@@ -33,25 +51,81 @@ const CollectionModal: React.FC<CollectionModalProps> = ({
 
   useEffect(() => {
     if (isEdit && collection) {
-      setValue("title", collection.title)
-      setValue("handle", collection.handle)
+      reset({
+        title: collection.title,
+        handle: collection.handle,
+      })
 
       if (collection.metadata) {
         Object.entries(collection.metadata).map(([key, value]) => {
-          const newMeta = metadata
-          newMeta.push({ key, value })
-          setMetadata(newMeta)
+          if (typeof value === "string") {
```

Comment:
The Metadata component only supports editing strings atm, so I just added this to prevent any breaking data types from being loaded in. As an example someone could have metadata: { factory_details: { country: "France", date: "2022-09-03", overseen_by: "Adrien" }} which would break the UI if we try to load it. We need to update the metadata component to handle more, if not all, types of data, but it's out of scope for this PR so I just added this safeguard in the meantime.

Reply:
Alright, got it, just wanted to understand the details since i am not very aware of all details in the admin 😂 sorry for the newby questions